pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,163 @@
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+type DiffOptions struct {
+	FilenameOptions    resource.FilenameOptions
+	Mapper             meta.RESTMapper
+	Typer              runtime.ObjectTyper
+	Namespace          string
+	EnforceNamespace   bool
+	DiffBufferModified []byte
+	DiffBufferOriginal []byte
+	Output             string
+	Codec              runtime.Encoder
+	Factory            cmdutil.Factory
+	Out                io.Writer
+	ErrOut             io.Writer
+	ext                string
+}
+
+func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	if o.Output != "yaml" && o.Output != "json" {
+		return fmt.Errorf("invalid output format %q, only yaml|json supported", o.Output)
+	}
+
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, o.EnforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Codec = runtime.NewCodec(f.JSONEncoder(), f.Decoder(true))
+	o.ext = "." + o.Output
+	o.Mapper = mapper
+	o.Typer = typer
+
+	r := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		o.DiffBufferModified, err = runtime.Encode(o.Codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+
+		// Verify the object exists in the cluster
+		if err := info.Get(); err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			}
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+
+		o.DiffBufferOriginal, err = kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+		if o.DiffBufferOriginal == nil {
+			return cmdutil.UsageErrorf(cmd, "no last-applied-configuration annotation found on resource: %s, to create the annotation, run the command with --create-annotation", info.Name)
+		}
+
+		//decode and delete the annotations fields
+		targetObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, o.DiffBufferOriginal)
+		if err != nil {
+			return nil
+		}
+		delete(targetObj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
```

Comment:
Followup https://github.com/kubernetes/kubernetes/pull/46560#discussion_r119524943.
Once again you can not delete all the annotations. The right thing to do here is deleting annotations only when it is all empty.

You can try the following config:
```
apiVersion: apps/v1beta1
kind: Deployment
metadata:
  annotations:
    hello: world
  name: nginx-deployment
spec:
  replicas: 3
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```
With your implementation, it shows the following even there is no change
```
3a4,5
>   annotations:
>     hello: world
```